zerver/views/user_settings.py

```diff
@@ -227,7 +227,7 @@ def json_change_settings(
             notification_sound, email_notifications_batching_period_seconds, default_language
         )
 
-    if new_password != "":
+    if new_password is not None:
```

Comment:
It looks like we already block `""` for a new password here, and that has a test:

```
def check_password_strength(password: str) -> bool:                                   
    """                                                                               
    Returns True if the password is strong enough,                                    
    False otherwise.                                                                  
    """                                                                               
    if len(password) < settings.PASSWORD_MIN_LENGTH:                                  
        return False                                                                  
                                                                                      
    if password == "":                                                                
        # zxcvbn throws an exception when passed the empty string, so                 
        # we need a special case for the empty string password here.                  
        return False                                                                  
```

So yeah, this is probably OK.